### ISSUE TITLE
Inaccurate grad scale with the new sds loss

### ISSUE THREAD

**Issue by HeliosZhao** [ashawkey/stable-dreamfusion#345]
### Description

Thanks a lot for your brilliant work.

Correct me if I am wrong. The grad scale of SDS loss should be multiplied by the loss instead of grad.

With the SDS loss formulated as `loss = 0.5 * F.mse_loss(latents.float(), targets, reduction='sum') / latents.shape[0]`, the grad scale should applied to the loss by `loss = loss * grad_scale`. Instead of multiply it on the grad before calculating the target ([here](https://github.com/ashawkey/stable-dreamfusion/blob/5550b91862a3af7842bb04875b7f1211e5095a63/guidance/sd_utils.py#L214)).

### Steps to Reproduce

NA

### Expected Behavior

NA

### Environment

NA

**Comment by HeliosZhao** [comment]
My fault. There is no difference here.
